src/Time/BoundaryHistory.hpp

```diff
@@ -473,6 +475,27 @@ std::ostream& BoundaryHistory<LocalVars, RemoteVars, CouplingResult>::print(
   return os;
 }
 
+template <typename LocalVars, typename RemoteVars, typename CouplingResult>
+std::string BoundaryHistory<LocalVars, RemoteVars, CouplingResult>::
+    pretty_print_without_data(const size_t padding_size) const {
```

Comment:
This is nearly identical to the preceding function.  Can you add a (template?) parameter to that one to control the data printing instead?